### PR TITLE
Feature/noamlandress/add cefama validation script

### PR DESCRIPTION
**Pull request by NoamLandress**
Required items, please complete
   
   Change(s):
   - Add new script for CEF AMA validation script

   Reason for Change(s):
   - As part of reaching CEF on top of AMA public preview, we need a CEF validation script for customers to run in order for them 
       to troubleshoot their environment.

   Version Updated:
   - This will be the first version

   Testing Completed:
   - Tested on all CEF supported operating systems

### CODE REVIEW

**Review by v-webr — APPROVED** [review]

File: DataConnectors/CEF/cef_AMA_troubleshoot.py [hunk 1]
```diff
@@ -0,0 +1,704 @@
...
+        self.verify_agent_is_running()
+        self.print_arc_version()
+        self.verify_oms_not_running()
+
+
+class DCRConfigurationVerifications:
+    '''
+    This class is for data collection rules verifications
+    '''
+    DCR_doc = "https://docs.microsoft.com/he-il/azure/azure-monitor/agents/data-collection-rule-overview"
```
Comment (v-webr) [inline DataConnectors/CEF/cef_AMA_troubleshoot.py:250]: See above comment

File: DataConnectors/CEF/cef_AMA_troubleshoot.py [hunk 2]
```diff
@@ -0,0 +1,704 @@
...
+    DCRA_doc = "https://docs.microsoft.com/en-us/rest/api/monitor/data-collection-rule-associations"
```
Comment (v-webr) [inline DataConnectors/CEF/cef_AMA_troubleshoot.py:251]: See above comment

File: DataConnectors/CEF/cef_AMA_troubleshoot.py [hunk 3]
```diff
@@ -0,0 +1,704 @@
...
+        if self.determine_Syslog_daemon():
+            self.verify_Syslog_daemon_listening()
+            self.verify_Syslog_daemon_forwarding_configuration()
+
+
+class OperatingSystemVerifications:
+    '''
+    This class is for general operating system verifications
+    '''
+    SELinux_documentation = "https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/using_selinux/changing-selinux-states-and-modes_using-selinux#changing-selinux-modes_changing-selinux-states-and-modes"
```
Comment (v-webr) [inline DataConnectors/CEF/cef_AMA_troubleshoot.py:422]: See above comment.

**Review by v-webr — CHANGES_REQUESTED** [review]

**Review by vmanojreddy — DISMISSED** [review]

File: DataConnectors/CEF/cef_AMA_troubleshoot.py [hunk 4]
```diff
@@ -0,0 +1,704 @@
...
+        '''
+        self.print_result_to_prompt()
+        self.log_result_to_file()
+
+
+class AgentInstallationVerifications:
+    '''
+    This class is for agent related verifications
+    '''
+    Agent_installation_doc = "https://docs.microsoft.com/he-il/azure/azure-monitor/agents/azure-monitor-agent-manage"
```
[the remark this answers is not in the document]
Comment (NoamLandress) [reply DataConnectors/CEF/cef_AMA_troubleshoot.py:182]: Done.

File: DataConnectors/CEF/cef_AMA_troubleshoot.py (hunk 3 above)
Comment (NoamLandress) [reply DataConnectors/CEF/cef_AMA_troubleshoot.py:422]: Done.

File: DataConnectors/CEF/cef_AMA_troubleshoot.py (hunk 2 above)
Comment (NoamLandress) [reply DataConnectors/CEF/cef_AMA_troubleshoot.py]: Resolved.

**Review by v-sabiraj — APPROVED** [review]

**Pull request merged**